### PR TITLE
Implement embedded infinity

### PR DESCRIPTION
**Pull request by Ma-cat**
### What problem does this PR solve?
* use nanobind to bind infinity cpp and python api
* direct pip install infinity to run infinity on the local file system
 
Issue link:#1181

### Type of change
- [x] New Feature (non-breaking change which adds functionality)
- [x] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [x] Refactoring
- [x] Test cases
- [x] Python SDK impacted, Need to update PyPI

### CODE REVIEW

**Comment by yuzhichang** [comment]
Remote and local test code can be combined into one.

File: third_party/nanobind/ext/robin_map/tests/CMakeLists.txt
Comment (JinHai-CN) [inline third_party/nanobind/ext/robin_map/tests/CMakeLists.txt:1]: robin_map/tests directory can be removed

File: third_party/nanobind/cmake/darwin-ld-cpython.sym
Comment (JinHai-CN) [inline third_party/nanobind/cmake/darwin-ld-cpython.sym:1]: can be removed.

File: third_party/nanobind/cmake/darwin-ld-pypy.sym
Comment (JinHai-CN) [inline third_party/nanobind/cmake/darwin-ld-pypy.sym:1]: can be removed

**Review by JinHai-CN — CHANGES_REQUESTED** [review]
Do we also need to build dockerfile?

**Pull request merged**